src/transsystem.c

```diff
@@ -45,18 +45,22 @@ callback_tree_before(T_PTR_tree entry) {
 
   info = (char*) tree_getinfo(entry);
   if (strcmp(info,"rules") == 0) {
+  	printf("Entering rules section\n");
     tree_brk_branch(entry);
     /* nbr_rules will be fill at this point */
-    rulescode_produce(entry, &_system);
+    rulescode_produce(entry, _system);
```

Comment:
In this line we don't need the '&' because the variable '_system' has been allocated by the build_problem_instance function.

I have not modified anything about the _init and _unsafe variables, so they are not initialized before calling these functions. 
In this case I think it's not a big problem because there will always be a 'target' and an 'init' section, so there won't be problems with the initialization of these variables.

Maybe it could be clearer to keep all the allocations together. If you think that this is better, I'll change it.